docs/GettingStartedAriesDev/PlugIns.md

```diff
@@ -0,0 +1,172 @@
+# Deeper Dive: Aca-Py Plug-Ins
+
+## What's in a Plug-In and How does it Work?
+
+Plug-ins are loaded on Aca-Py startup based on the following parameters:
+
+* `--plug-in` - identifies the plug-in library to load
+* `--block-plugin` - identifies plug-ins (includign built-ins) that are *not* to be loaded
+* `--plugin-config` - identify a configuration parameter for a plug-in
+* `--plugin-config-value` - identify a *value* for a plug-in configuration
+
+
+The `--plug-in` parameter specifies a package that is loaded by Aca-Py at runtime, and extends Aca-Py by adding support for additional protocols and message types, and/or extending the Admin API with additional endpoints.
+
+The original plug-in design (which we will call the "old" model) explicitely indluded `message_types.py` `routes.py` (to add Admin API's).  But functionality was added later (we'll call this the "new" model) to allow the plug-in to include a generic `setup` package that could perform arbitrary initialization.  The "new" model also includes support for a `definition.py` file that can specify plug-in version information  (major/minor plug-in version, as well as the minimum supported version (if another agent is running an older version of the plug-in)).
+
+### setup.py
+
+If a setup method is provided, it will be called.  If not, the `message_types.py` and `routes.py` will be explicitely loaded.
```

Comment:
should we mention that it would be in the package/module `__init__.py`?

```
async def setup(context: InjectionContext):
    pass
```

Reply:
Good point I'll do that